mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2162,27 +2162,48 @@ where
 /// It's used to verify that all the inputs of the taker-sent dex fee are signed/owned by the taker's pubkey.
 /// It's used also by watcher to verify that all the taker payment inputs are signed/owned by the taker's pubkey.
 /// The `expected_pub` should be the taker's pubkey in compressed (33-byte) format.
-pub fn check_all_utxo_inputs_signed_by_pub<T: UtxoCommonOps>(
+pub async fn check_all_utxo_inputs_signed_by_pub<T: UtxoCommonOps>(
     coin: &T,
     tx: &UtxoTx,
     expected_pub: &[u8],
 ) -> Result<bool, MmError<ValidatePaymentError>> {
     let expected_pub =
         H264::from_slice(expected_pub).map_to_mm(|e| ValidatePaymentError::TxDeserializationError(e.to_string()))?;
     let mut unsigned_tx: TransactionInputSigner = tx.clone().into();
-    unsigned_tx.consensus_branch_id = coin.as_ref().conf.consensus_branch_id;
 
     for (idx, input) in tx.inputs.iter().enumerate() {
         let script = Script::from(input.script_sig.clone());
         let pubkey = if input.has_witness() {
             // Extract the pubkey from a P2WPKH scriptSig.
             pubkey_from_witness_script(&input.script_witness).map_to_mm(ValidatePaymentError::TxDeserializationError)?
         } else if does_script_spend_p2pk(&script) {
-            // For P2PK scriptsSigs, verfiy that the signature corresponds to the expected public key.
+            // If the transaction is overwintered, we need to set the consensus branch id and the input's amount.
+            // This is needed for the sighash calculation.
+            if unsigned_tx.overwintered {
+                let prev_output_tx_hash = input.previous_output.hash.reversed().into();
+                let prev_output_index = input.previous_output.index as usize;
+                let prev_tx = coin
+                    .as_ref()
+                    .rpc_client
+                    .get_verbose_transaction(&prev_output_tx_hash)
+                    .compat()
+                    .await
+                    .map_err(|e| ValidatePaymentError::TxDeserializationError(format!("Failed to get prev tx: {e}")))?;
+                let prev_tx: UtxoTx = deserialize(prev_tx.hex.0.as_slice())?;
```

Comment:
Why use `get_verbose_transaction` if we are gonna deserialize the hex anyways?

Reply:
reverted back to amount + consensus based verification here: [86b0949](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2408/commits/86b0949d037108bbe6917118a0fa0875af92dedc)
replaced verbose tx with tx bytes here: [95411c0](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2408/commits/95411c034218c7367b08753ae7098c9893f280e1)